<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.550">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Piero Palacios Bernuy">
<meta name="dcterms.date" content="2024-02-23">
<meta name="keywords" content="450k array, methylation, CpG, CpGs islands">

<title>Analysis of Methylated DNA</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<style>html{ scroll-behavior: smooth; }</style>
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>


<meta name="citation_title" content="Analysis of Methylated DNA">
<meta name="citation_abstract" content="This document is part of a series of the analysis of Omics data. Especifycally, here is showed how to analyze Illumina 450k array and Methyl-Seq data with Bioconductor packages. Also, it&amp;amp;#039;s showcased how to make plots of the methylation data in the context of genomic positions and genes strutctures.
">
<meta name="citation_keywords" content="450k array,methylation,CpG,CpGs islands">
<meta name="citation_author" content="Piero Palacios Bernuy">
<meta name="citation_publication_date" content="2024-02-23">
<meta name="citation_cover_date" content="2024-02-23">
<meta name="citation_year" content="2024">
<meta name="citation_online_date" content="2024-02-23">
<meta name="citation_language" content="en">
<meta name="citation_journal_title" content="An open source portfolio">
<meta name="citation_reference" content="citation_title=BSmooth: From whole genome bisulfite sequencing reads to differentially methylated regions;,citation_author=Kasper D. Hansen;,citation_author=Benjamin Langmead;,citation_author=Rafael A. Irizarry;,citation_publication_date=2012;,citation_cover_date=2012;,citation_year=2012;,citation_doi=10.1186/gb-2012-13-10-r83;,citation_volume=13;">
<meta name="citation_reference" content="citation_title=Preprocessing, normalization and integration of the Illumina HumanMethylationEPIC array with minfi;,citation_abstract=The minfi package is widely used for analyzing Illumina DNA methylation array data. Here we describe modifications to the minfi package required to support the HumanMethylationEPIC (“EPIC”) array from Illumina. We discuss methods for the joint analysis and normalization of data from the HumanMethylation450 (“450k”) and EPIC platforms. We introduce the single-sample Noob (ssNoob) method, a normalization procedure suitable for incremental preprocessing of individual methylation arrays and conclude that this method should be used when integrating data from multiple generations of Infinium methylation arrays. We show how to use reference 450k datasets to estimate cell type composition of samples on EPIC arrays. The cumulative effect of these updates is to ensure that minfi provides the tools to best integrate existing and forthcoming Illumina methylation array data.The minfi package version 1.19.12 or higher is available for all platforms from the Bioconductor project.Supplementary data are available at Bioinformatics online.;,citation_author=Jean-Philippe Fortin;,citation_author=Jr Triche;,citation_author=Kasper D Hansen;,citation_publication_date=2016-11;,citation_cover_date=2016-11;,citation_year=2016;,citation_fulltext_html_url=https://doi.org/10.1093/bioinformatics/btw691;,citation_issue=4;,citation_doi=10.1093/bioinformatics/btw691;,citation_issn=1367-4803;,citation_volume=33;,citation_journal_title=Bioinformatics;">
</head>

<body>

<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Analysis of Methylated DNA</h1>
          </div>

    
    <div class="quarto-title-meta-container">
      <div class="quarto-title-meta-column-start">
        
        <div class="quarto-title-meta">

                <div>
            <div class="quarto-title-meta-heading">Author</div>
            <div class="quarto-title-meta-contents">
                        <p>Piero Palacios Bernuy <a href="mailto:p.palacios.bernuy@gmail.com" class="quarto-title-author-email"><i class="bi bi-envelope"></i></a> <a href="https://orcid.org/0000-0001-6729-4080" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
                      </div>
          </div>
                
                <div>
            <div class="quarto-title-meta-heading">Published</div>
            <div class="quarto-title-meta-contents">
              <p class="date">February 23, 2024</p>
            </div>
          </div>
          
                
              </div>
      </div>
      <div class="quarto-title-meta-column-end quarto-other-formats-target">
      </div>
    </div>

    <div>
      <div class="abstract">
        <div class="block-title">Abstract</div>
        <p>This document is part of a series of the analysis of Omics data. Especifycally, here is showed how to analyze Illumina 450k array and Methyl-Seq data with Bioconductor packages. Also, it’s showcased how to make plots of the methylation data in the context of genomic positions and genes strutctures.</p>
      </div>
    </div>

    <div>
      <div class="keywords">
        <div class="block-title">Keywords</div>
        <p>450k array, methylation, CpG, CpGs islands</p>
      </div>
    </div>

    <div class="quarto-other-links-text-target">
    </div>  </div>
</header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#methylation-analysis" id="toc-methylation-analysis" class="nav-link active" data-scroll-target="#methylation-analysis"><span class="header-section-number">1</span> Methylation analysis</a>
  <ul class="collapse">
  <li><a href="#methylation" id="toc-methylation" class="nav-link" data-scroll-target="#methylation"><span class="header-section-number">1.1</span> Methylation</a></li>
  <li><a href="#cpgs" id="toc-cpgs" class="nav-link" data-scroll-target="#cpgs"><span class="header-section-number">1.2</span> CpG’s</a></li>
  <li><a href="#bisulfite-treatment" id="toc-bisulfite-treatment" class="nav-link" data-scroll-target="#bisulfite-treatment"><span class="header-section-number">1.3</span> Bisulfite treatment</a></li>
  <li><a href="#methylation-450k-array" id="toc-methylation-450k-array" class="nav-link" data-scroll-target="#methylation-450k-array"><span class="header-section-number">1.4</span> Methylation 450k array</a></li>
  <li><a href="#methylation-450k-array---multi-resolution-analysis" id="toc-methylation-450k-array---multi-resolution-analysis" class="nav-link" data-scroll-target="#methylation-450k-array---multi-resolution-analysis"><span class="header-section-number">1.5</span> Methylation 450k array - Multi-resolution Analysis</a></li>
  <li><a href="#methyl-seq" id="toc-methyl-seq" class="nav-link" data-scroll-target="#methyl-seq"><span class="header-section-number">1.6</span> Methyl-Seq</a></li>
  </ul></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
<div class="quarto-alternate-notebooks"><h2>Notebooks</h2><ul><li><a href="notebooks/Methylation_multiresolution_analysis-preview.html"><i class="bi bi-journal-code"></i>DNA methylation measurement</a></li><li><a href="notebooks/Methylation_analysis-preview.html"><i class="bi bi-journal-code"></i>DNA methylation measurement</a></li><li><a href="notebooks/Methyl_Seq-preview.html"><i class="bi bi-journal-code"></i>DNA methylation measurement by sequencing</a></li></ul></div></nav>
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">



  


<section id="methylation-analysis" class="level2" data-number="1">
<h2 data-number="1" class="anchored" data-anchor-id="methylation-analysis"><span class="header-section-number">1</span> Methylation analysis</h2>
<section id="methylation" class="level3" data-number="1.1">
<h3 data-number="1.1" class="anchored" data-anchor-id="methylation"><span class="header-section-number">1.1</span> Methylation</h3>
<p>Epigenetics is the study of how your behaviors and environment can cause changes that affect the way your genes work. Unlike genetic changes, epigenetic changes are reversible and do not change your DNA sequence, but they can change how your body reads a DNA sequence (CDC,2024).</p>
<p>Talking about the genes, these can only be expressed when the chromatin is in it’s relaxed form (euchromatin) and reversely, these cannot be expressed in it’s compacted form (hetero-chromatin). Methylation of cytosine’s control part of this process of the chromatin and it’s really important to understand is a gene (or the surroundings) is methylated.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="https://static.cambridge.org/binary/version/id/urn:cambridge.org:id:binary:20220108070724404-0024:S0924270821000181:S0924270821000181_fig3.png?pub-status=live" class="lightbox" data-gallery="quarto-lightbox-gallery-1" data-glightbox="description: .lightbox-desc-1" title="DNA Methylation"><img src="https://static.cambridge.org/binary/version/id/urn:cambridge.org:id:binary:20220108070724404-0024:S0924270821000181:S0924270821000181_fig3.png?pub-status=live" class="img-fluid figure-img" width="447" alt="DNA Methylation"></a></p>
<figcaption>DNA Methylation</figcaption>
</figure>
</div>
</section>
<section id="cpgs" class="level3" data-number="1.2">
<h3 data-number="1.2" class="anchored" data-anchor-id="cpgs"><span class="header-section-number">1.2</span> CpG’s</h3>
<p>The cytosine’s (C) that are methylated are followed by a guanine (G). This pattern is called CpG in the context of epigenetics. These CpG’s tend to cluster into groups (CpG islands) and, these Cpg islands tend to be close to the promoter region of the genes.</p>
</section>
<section id="bisulfite-treatment" class="level3" data-number="1.3">
<h3 data-number="1.3" class="anchored" data-anchor-id="bisulfite-treatment"><span class="header-section-number">1.3</span> Bisulfite treatment</h3>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="https://www.diagenode.com/img/categories/bisulfite-conversion/bisulfite.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2" data-glightbox="description: .lightbox-desc-2" title="Bisulfite Treatment"><img src="https://www.diagenode.com/img/categories/bisulfite-conversion/bisulfite.png" class="img-fluid figure-img" alt="Bisulfite Treatment"></a></p>
<figcaption>Bisulfite Treatment</figcaption>
</figure>
</div>
<p>If a C is methylated the bisulfite treatment won´t change the cytosine but, if the C is not methylated, the C will suffer a transition to a T.</p>
</section>
<section id="methylation-450k-array" class="level3" data-number="1.4">
<h3 data-number="1.4" class="anchored" data-anchor-id="methylation-450k-array"><span class="header-section-number">1.4</span> Methylation 450k array</h3>
<p>To see how to analyze from a Illumina 450k array with the <span class="citation" data-cites="minfi">(<a href="#ref-minfi" role="doc-biblioref">Fortin, Triche, and Hansen 2016</a>)</span> package, click to the script below the image.</p>
<div class="quarto-embed-nb-cell">
<div id="cell-fig-meth" class="cell">
<div class="cell-output cell-output-display">
<div id="fig-meth" class="quarto-figure quarto-figure-center quarto-float anchored" width="672" height="480">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-meth-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/notebooks-Methylation_analysis-fig-meth-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3" data-glightbox="description: .lightbox-desc-3"><img src="index_files/figure-html/notebooks-Methylation_analysis-fig-meth-output-1.png" id="fig-meth" width="672" height="480" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-meth-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1
</figcaption>
</figure>
</div>
</div>
</div>
<a class="quarto-notebook-link" id="nblink-1" href="notebooks/Methylation_analysis-preview.html#cell-fig-meth">Source: DNA methylation measurement</a></div>
</section>
<section id="methylation-450k-array---multi-resolution-analysis" class="level3" data-number="1.5">
<h3 data-number="1.5" class="anchored" data-anchor-id="methylation-450k-array---multi-resolution-analysis"><span class="header-section-number">1.5</span> Methylation 450k array - Multi-resolution Analysis</h3>
<p>To see how to analyze at a multi-resolution level from a Illumina 450k array with the <span class="citation" data-cites="minfi">(<a href="#ref-minfi" role="doc-biblioref">Fortin, Triche, and Hansen 2016</a>)</span> package, click to the script below the table.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="notebooks/track_plot.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4" data-glightbox="description: .lightbox-desc-4" title="Methylation DMR"><img src="notebooks/track_plot.png" class="img-fluid figure-img" alt="Methylation DMR"></a></p>
<figcaption>Methylation DMR</figcaption>
</figure>
</div>
<div class="quarto-embed-nb-cell">
<div id="cell-Island-status" class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>mcols(granges(dat$object))$type <span class="op">|&gt;</span> </span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>    table() <span class="op">|&gt;</span> </span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>    prop.table() <span class="op">|&gt;</span> </span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>    <span class="im">as</span>.data.frame() <span class="op">|&gt;</span> </span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>    dplyr::rename(<span class="st">"Island Status"</span> <span class="op">=</span> Var1) <span class="op">|&gt;</span> </span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>    gt::gt()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="island-status" class="cell-output cell-output-display">

</div>
</div>
<a class="quarto-notebook-link" id="nblink-2" href="notebooks/Methylation_multiresolution_analysis-preview.html#cell-Island-status">Source: DNA methylation measurement</a></div>
</section></section><div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" id="quarto-citation"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@online{palacios bernuy2024,
  author = {Palacios Bernuy, Piero},
  title = {Analysis of {Methylated} {DNA}},
  date = {2024-02-23},
  langid = {en},
  abstract = {This document is part of a series of the analysis of Omics
    data. Especifycally, here is showed how to analyze Illumina 450k
    array and Methyl-Seq data with Bioconductor packages. Also, it’s
    showcased how to make plots of the methylation data in the context
    of genomic positions and genes strutctures.}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-palacios bernuy2024" class="csl-entry quarto-appendix-citeas" role="listitem">
Palacios Bernuy, Piero. 2024. <span>“Analysis of Methylated DNA.”</span>
An Open Source Portfolio. February 23, 2024.
</div></div></section></div></main></div>

<section id="methyl-seq" class="level3" data-number="1.6">
<h3 data-number="1.6" data-anchor-id="methyl-seq"><span class="header-section-number">1.6</span> Methyl-Seq</h3>
<p>Analyzing bisulfite sequencing data requires a server or an HPC. U can use the <em>methylseq</em> pipeline from <a href="https://nf-co.re/methylseq/2.6.0">nfcore</a> to get the .cov files to proced with the downstream analysis on R.</p>
<p>To see how to analyze Whole-genome bisulfite sequencing (WGBS) with the <span class="citation" data-cites="bsseq">(<a href="#ref-bsseq" role="doc-biblioref">Hansen, Langmead, and Irizarry 2012</a>)</span>, click on the script below the image.</p>
<section id="wgbs" class="level4" data-number="1.6.1">
<h4 data-number="1.6.1" data-anchor-id="wgbs"><span class="header-section-number">1.6.1</span> WGBS</h4>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="notebooks/track_methyl_seq_plot.png" class="lightbox" data-gallery="quarto-lightbox-gallery-5" data-glightbox="description: .lightbox-desc-5" title="Methylation DMR"><img src="notebooks/track_methyl_seq_plot.png" class="img-fluid figure-img" width="523" alt="Methylation DMR"></a></p>
<figcaption>Methylation DMR</figcaption>
</figure>
</div>
<div class="quarto-embed-nb-cell">
<div id="cell-fig-dmr" class="cell">
<div class="cell-output cell-output-display">
<div id="fig-dmr" class="quarto-figure quarto-figure-center quarto-float" width="672" height="480">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-dmr-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="index_files/figure-html/notebooks-Methyl_Seq-fig-dmr-output-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-6" data-glightbox="description: .lightbox-desc-6"><img src="index_files/figure-html/notebooks-Methyl_Seq-fig-dmr-output-1.png" id="fig-dmr" width="672" height="480" class="figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-dmr-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2
</figcaption>
</figure>
</div>
</div>
</div>
<a class="quarto-notebook-link" id="nblink-3" href="notebooks/Methyl_Seq-preview.html#cell-fig-dmr">Source: DNA methylation measurement by sequencing</a></div>
</section>
</section>

<section id="references" class="level2 unnumbered">
<h2 class="unnumbered" data-anchor-id="references">References</h2>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-minfi" class="csl-entry" role="listitem">
Fortin, Jean-Philippe, Jr Triche Timothy J., and Kasper D Hansen. 2016. <span>“<span class="nocase">Preprocessing, normalization and integration of the Illumina HumanMethylationEPIC array with minfi</span>.”</span> <em>Bioinformatics</em> 33 (4): 558–60. <a href="https://doi.org/10.1093/bioinformatics/btw691">https://doi.org/10.1093/bioinformatics/btw691</a>.
</div>
<div id="ref-bsseq" class="csl-entry" role="listitem">
Hansen, Kasper D., Benjamin Langmead, and Rafael A. Irizarry. 2012. <span>“<span></span>BSmooth: From Whole Genome Bisulfite Sequencing Reads to Differentially Methylated Regions<span></span>”</span> 13: R83. <a href="https://doi.org/10.1186/gb-2012-13-10-r83">https://doi.org/10.1186/gb-2012-13-10-r83</a>.
</div>
</div>
<div class="hidden" aria-hidden="true">
<span class="glightbox-desc lightbox-desc-1">DNA Methylation</span>
<span class="glightbox-desc lightbox-desc-2">Bisulfite Treatment</span>
<span class="glightbox-desc lightbox-desc-3">Figure&nbsp;1: </span>
<span class="glightbox-desc lightbox-desc-4">Methylation DMR</span>
<span class="glightbox-desc lightbox-desc-5">Methylation DMR</span>
<span class="glightbox-desc lightbox-desc-6">Figure&nbsp;2: </span>
</div>
</section>


<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
 <!-- /content -->
<script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","selector":".lightbox","openEffect":"zoom","loop":false,"descPosition":"bottom"});
window.onload = () => {
  lightboxQuarto.on('slide_before_load', (data) => {
    const { slideIndex, slideNode, slideConfig, player, trigger } = data;
    const href = trigger.getAttribute('href');
    if (href !== null) {
      const imgEl = window.document.querySelector(`a[href="${href}"] img`);
      if (imgEl !== null) {
        const srcAttr = imgEl.getAttribute("src");
        if (srcAttr && srcAttr.startsWith("data:")) {
          slideConfig.href = srcAttr;
        }
      }
    } 
  });

  lightboxQuarto.on('slide_after_load', (data) => {
    const { slideIndex, slideNode, slideConfig, player, trigger } = data;
    if (window.Quarto?.typesetMath) {
      window.Quarto.typesetMath(slideNode);
    }
  });

};
          </script>




<script src="site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>